logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageContent.kt

```diff
@@ -1,18 +1,30 @@
 package com.wire.kalium.logic.data.message
 
+import com.wire.kalium.logic.data.conversation.Member
 import com.wire.kalium.protobuf.messages.QualifiedConversationId
 
 sealed class MessageContent {
-    data class Text(val value: String) : MessageContent()
-    data class Calling(val value: String) : MessageContent()
-    data class Asset(val value: AssetContent) : MessageContent()
-    data class DeleteMessage(val messageId: String) : MessageContent()
-    data class TextEdited(val editMessageId :String, val newContent : String) : MessageContent()
+
+    sealed class Client : MessageContent()
+    sealed class Server : MessageContent()
+
+    // client message content types
+    data class Text(val value: String) : Client()
+    data class Calling(val value: String) : Client()
+    data class Asset(val value: AssetContent) : Client()
+    data class DeleteMessage(val messageId: String) : Client()
+    data class TextEdited(val editMessageId :String, val newContent : String) : Client()
     data class DeleteForMe(
         val messageId: String,
         val conversationId: String,
         val qualifiedConversationId: QualifiedConversationId?
-    ) : MessageContent()
+    ) : Client()
```

Comment:
Shouldn't we nest these inside the `Client` sealed class in order to help compiler be exhaustive on possible `when` clauses?

Reply:
I think from compiler's perspective it’s no difference, it still knows which are extensions of which class so it’s mainly a matter of how we see / format it. With having all items inside Client/Server we would need to change many lines just for it to be for instance `MessageContent.Client.Text` and we usually don’t need such detailed info in every place, so current way I think is the best of both worlds: we have hierarchy and all advantages of sealed classes without the need of changing every line of code and having overwhelming names. What do you think?

Reply:
I said it because I saw multiple times, IDE warning around non exhaustive `when` clauses even though they were. And I thought it was because of that. But if you say it is not that's fine with me too. It's also true that without the extra nested class we gain in flexibility and lack of pain 😅